### ISSUE TITLE
loading within a ArcGIS Web AppBuilder widget

### ISSUE THREAD

**Issue by canadajebus** [spatialillusions/milsymbol#105]
Hi, I'm trying to implement milsymbol within a custom widget used in ArcGIS Web AppBuilder (Developer Edition).

I am using the examples similar to your 3d arcgis example, but am getting the error ms.Symbol is not a constructor.  I believe this is likely caused by the way the DOJO AMD loader stuff works.

Is there a solution for this?

**Comment by spatialillusions** [comment]
I haven't used ArcGIS Web AppBuilder, but if you check the documentation for milsymbol you can see that there are two different ways for loading milsymbol into dojo.

From the documentation:

-------------------------------

In a framework like [Dojo](https://dojotoolkit.org) you can load milsymbol like this: 

```html
<html>
  <body>
    <script src="dojo/dojo.js" data-dojo-config="async: true"></script>
    <script src="app/milsymbol.js"></script>
    <script>
      require([
        "milsymbol"
      ], function(ms){
        var symbol = new ms.Symbol('SFG-UCI----D',{size:30});
      });
    </script>
  </body>
</html>
```

This way of loading milsymbol makes it possible to use Dojo from an external source, like js.arcgis.com. 

If you only are using it with local sources, you can set `umdNamedDefine:false` in *webpack.config.js* and rebuild it by running `npm run build`, and now you can `require(["app/milsymbol"].....` if *milsymbol.js* is placed in your app folder. See more about building under Building milsymbol.

-----------------------------------------

I guess that you have tried the first way, but it might be possible that you will have to move to the other way where you set umdNamedDefine to false and make a custom build. 
Also, have you remembered to add `ms` to the function call? I'm asking because I know I have forgotten this some times.

**Comment by canadajebus** [comment]
The worked using the npm rebuild.

Thanks for the quick response and all the effort you've put into this library!

I had built an arcgis js widget with your library back when it was v0.5.5b and it worked great in our apps.  Over the last year we've migrated all of our apps web appbuilder versions and have yet to replace the functionality.  I'll try to post it to my github in the future as well.

**Comment by spatialillusions** [comment]
Could you make a test for me?

If you set 
libraryTarget: "amd"
but keep 
umdNamedDefine: true
will this build work for you?

I'm trying to figure out the best way to implement #106.

**Comment by canadajebus** [comment]
The build works, but using the new milsymbol.js in my widget causes TypeError: ms.Symbol is not a contructor again.

If you have the ArcGIS Web AppBuilder Developer Edition I can provide you a sample of the widget I'm developing (by email) right away.  Its still early on but enough is done to test.  I'll post a more full version once its ready, hopefully next week.

**Comment by spatialillusions** [comment]
Good to know. I got an idea of how to implement something around this during this week that will simplify how to get an AMD build, should just be a few lines of code.

Unfortunatly I don't have ArcGIS Web AppBuilder Developer Edition. But I have some Esri contacts that are quite interested in milsymbol, so I'll get in touch with them as soon as I have a solution in place and we might work something out.

**Comment by spatialillusions** [comment]
I have now implemented #106, and it should now be possible to run `npm run build-amd` to make an AMD only module. 
Please let me know if there are any more issues.
